python/ray/data/tests/test_dataset.py

```diff
@@ -2571,6 +2582,40 @@ def __call__(self, x):
     assert values == [11, 15, 19]
 
 
+def test_map_batches_generator(ray_start_regular_shared, tmp_path):
+    ctx = DatasetContext.get_current()
+    ctx.execution_options.preserve_order = True
```

Comment:
can we check sorted output via `sorted(...)`, instead of setting the option to preserve order here? o.w. we are at risk not test our default behavior.

Reply:
made the change, but `preserve_order=True` is what we set for the other `map_batches` tests

Reply:
Thanks @amogkam, we should clean them up in the future...